docs/task-files.md

```diff
@@ -414,3 +416,68 @@ Include:
   Parameters:
     resourcePrefix: my
 ```
+
+## Templating
+Org-formation supports the [nunjucks](https://mozilla.github.io/nunjucks/) template engine to generate cloudformation
+templates from nunjucks based templates.  Only files with a `.nj` extension will be processed as nunjucks
+templates.
+
+### Example:
+Assume we want to create one security group that allows access from multiple ingress ports.
+
+security-group.nj:
+```
+Description: Nunjucks Security group template
+AWSTemplateFormatVersion: 2010-09-09
+Resources:
+  SecurityGroup:
+    Type: 'AWS::EC2::SecurityGroup'
+    Properties:
+      GroupDescription: "Open ports for incoming traffic"
+      VpcId: "vpc-1234ABC"
+      SecurityGroupIngress:
+{% for port in Data.ports %}
+        - CidrIp: "0.0.0.0/0"
+          FromPort: {{ port }}
+          ToPort: {{ port }}
+          IpProtocol: tcp
+{% endfor %}
+```
+
+Deploy with [update-stacks](#update-stacks) and pass in port values with `Data`:
+```
+SecurityGroupExample:
+  Type: update-stacks
+  Template: ./security-group.nj
+  StackName: SecurityGroupExample
+  Data:
```

Comment:
IMO `TextTemplatingContext` is a little too long because it will need to be referenced in templated files.  How about we make it more specific name to make it easier to identify with the code supporting the functionality?  how about just `nunjucks`?

Reply:
This line has no influence on how the contextual property that is accessed from within the template is called. if we call it TextTemplatingContext here, the way you access it from within the template would still be `Data`, because of [this line](https://github.com/zaro0508/org-formation-cli/blob/6c620db9abad8494c63058182221b4b4253663c9/src/yaml-cfn/index.ts#L129).

should we inject it as `Context` (instead of Data)?
i agree that injecting it as `TextTemplatingContext` would definitely be too much

Reply:
Using `Context` sounds good to me.

when i say `..easier to identify with code..` I just meant that searching thru the code, examples, and docs I can just search for `nunjucks` instead of multiple different concepts to locate all things around this feature.

Reply:
sorry, i didn't pick up on the second part of your suggestion.

my thinking:
The feature is 'text templating', 'nunjucks' is an implementation of this. just like jinja (or something else) could be. Maybe at a later date there could be different dialects or we might swap out nunjucks for jinja. i do agree 'text templating' is a mouth full..

The object provided in the settings is the contextual object used for text templating.
In the task file just calling it Context i would fear is confusing. it could be the context of many things. therefore i think it makes sense to call it TextTemplatingContext. in the template calling it Context does make sense.

Reply:
agreed.  will use `Context` in the ofn template and use `TextTemplatingContext` in the code.

Reply:
refactored to use `TemplatingContext` in both ofn task and cfn template files.